### ISSUE TITLE
unittest.TestCase.assertRegexpMatches() renamed to unittest.TestCase.assertRegex() in Python 3.2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8372]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8372|
|Type|enhancement|
|Created|2016-06-01 09:21:49Z|
|Branch|https://github.com/twisted/twisted/tree/8372-rodrigc-assertRegexp|

TestCase.assertRegexpMatches() was introduced in Python 2.7:
https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.assertRegexpMatches
    
but renamed to TestCase.assertRegex() in Python 3.2:
https://docs.python.org/3.2/library/unittest.html#unittest.TestCase.assertRegex

so testcases which use assertRegexpMatches() break on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8372 8372
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8372_rodrigc_assertRegexp 8372-rodrigc-assertRegexp
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__None None
time__1464772909088136 1464772909088136
changetime__1465639688161352 1465639688161352
version__None None
owner__Amber_Brown__HawkOwl___hawkowl_____ Amber Brown (HawkOwl) <hawkowl@...>
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/89

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for the py3 help, rodrigc.

Mutating global state in other libraries is a generally bad practice in Python overall.  This means that another project with, say, an optional dependency on Twisted, might accidentally depend on `unittest.TestCase.assertRegex` and then break if Twisted wasn't installed.

We already have a `TestCase` subclass we use everywhere - `twisted.trial.unittest.SynchronousTestCase` - so you can add the method to that.

Please address the feedback and resubmit.  Thank you!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set owner to @glyph|
|-|-|

OK, I uploaded a new patch to:

https://github.com/twisted/twisted/pull/89

and added assertRegex() to class _Assertions

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

I saw that Glyphs comments were based on https://github.com/twisted/twisted/pull/89 submitted 8 days ago, but I can not see that code.

What is the diff between the current version and the version reviewed by Glyph?

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc commented|
|-|-|

Sorry for not keeping the old patch.  The old patch did not modify *class _Assertions* and just did:

```
if sys.version_info < (3, 2):
    setattr(unittest.TestCase, "assertRegex", unittest.TestCase.assertRegexpMatches)
```

which glyph did not like.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Looks good. Thanks!

I have added the docstring to the new method.

Waiting for build results and will merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban removed owner|
|-|-|

Tests look good, but I think that this should be a feature and not a misc.

I have updated the news fragment and I am putting it back for review.

Please check that changes make sense.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset e8965c3fc2d3317dac538d5e94127dfb6c22e61a
```
#!CommitTicketReference repository="" revision="e8965c3fc2d3317dac538d5e94127dfb6c22e61a"
Merge 8372-rodrigc-assertRegexp: Replace unittest.TestCase.assertRegexpMatches() with unittest.TestCase.assertRegex() for Py2/3 compat

Authors: rodrigc, adiroiban
Reviewer: hawkowl
Fixes: #8372
```
